ast/compile_test.go

```diff
@@ -109,6 +110,122 @@ func TestCompilerExample(t *testing.T) {
 	assertNotFailed(t, c)
 }
 
+func TestCompilerUserFunctionConflicts(t *testing.T) {
```

Comment:
High-level comment: without a "note" or descriptive error message, it's hard to understand what each of these test cases is exercising.